src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerOptionsExtensions.cs

```diff
@@ -17,7 +17,7 @@ internal static class AnalyzerOptionsExtensions
 
         private const string DotnetAnalyzerDiagnosticSeverityKey = DotnetAnalyzerDiagnosticPrefix + "." + SeveritySuffix;
 
-        private static string GetCategoryBasedDotnetAnalyzerDiagnosticSeverityKey(string category)
+        internal static string GetCategoryBasedDotnetAnalyzerDiagnosticSeverityKey(string category)
```

Comment:
@Youssef1313 I believe a better approach here would be to cache this onto a static concurrent dictionary. We do not expect more than a handful distinct diagnostic categories, so the number of entries for the map would be limited. Having a static cache preserves them across compilations, which will provide additional savings in the IDE live analysis scenario.

Reply:
I see that you did mention the static dictionary approach in the associated issue description. Was there a specific reason you went with the current approach?